[MWAN] ICMP and CF Tunnel (#8809)
cloudflare/cloudflare-docs: content/magic-wan/how-to/traceroute.md

---
pcx_content_type: how-to
title: Run traceroute
meta:
    description: Learn what settings you need to change to perform a useful `traceroute` to an endpoint behind a Cloudflare Tunnel.
---

# Run `traceroute`

> If you have a Magic WAN client connected through [GRE, IPsec](/magic-wan/get-started/configure-tunnels/), [CNI](/network-interconnect/) or [WARP](/magic-wan/tutorials/warp/) and want to perform a `traceroute` to an endpoint behind a [Cloudflare Tunnel](/cloudflare-one/connections/connect-apps/), there are a couple of settings you need to have in mind for the command to return useful information.

Reviewer: ```suggestion
If you have a Magic WAN client connected through [GRE, IPsec](/magic-wan/get-started/configure-tunnels/), [CNI](/network-interconnect/) or [WARP](/magic-wan/tutorials/warp/) and want to perform a `traceroute` to an endpoint behind a [Cloudflare Tunnel](/cloudflare-one/connections/connect-apps/), the following settings must be applied for the command to return useful information.
```
